src/ILVerify/tests/ILMethodTester.cs

```diff
@@ -0,0 +1,208 @@
+    /// Describes a test case with a method that contains invalid IL with the expected VerifierErrors
+    /// </summary>
+    class InvalidILTestCase : TestCase
+    {
+        public List<VerifierError> ExpectedVerifierError { get; set; }
+
+        /// <summary>
+        /// The expected exception during validation.
+        /// If no exception is thrown during verification this must be null.
+        /// </summary>
+        public bool ExpectsException { get; set; }
+    }
+
+    /// <summary>
+    /// Parses the methods in the test assemblies. 
+    /// It loads all assemblies from the test folder defined in <code>TestDataLoader.TESTASSEMBLYPATH</code>
+    /// This class feeds the xunit Theories
+    /// </summary>
+    class ILReader
+    {
+        public enum TestMethodType
+        {
+            Valid, Invalid
+        }
+
+        /// <summary>
+        /// Returns all methods that contain valid IL code based on the following naming convention:
+        /// [FriendlyName]_Valid
+        /// The method must contain 1 '_'. The part before the '_' is a friendly name describing what the method does. 
+        /// The word after the '_' has to be 'Valid'. 
+        /// E.g.: 'SimpleAdd_Valid'
+        /// </summary>
+        public static IEnumerable<Object[]> GetMethodsWithValidIL()
+        {
+            var methodSelector = new Func<string[], EcmaMethod, EcmaMethodIL, TestCase>((mparams, method, methodIL) =>
+            {
+                if (mparams.Length == 2 && mparams[1].ToLower() == "valid")
+                {
+                    return new ValidILTestCase { ILImporter = new ILImporter(method, methodIL) };
+                }
+                return null;
+            });
+            return GetTestMethodsFromDll(methodSelector);
+        }
+
+        /// <summary>
+        /// Returns all methods that contain valid IL code based on the following naming convention:
+        /// [FriendlyName]_Invalid_[ExpectedVerifierError1].[ExpectedVerifierError2]_[ExpectedExceptionDuringValidation]
+        /// The method name must contain 4 '_' characters.
+        /// 1. part: a friendly name
+        /// 2. part: must be the word 'Invalid'
+        /// 3. part: the expected VerifierErrors as string separated by '.'.
+        /// 4. part: Either 'ExpectsException' or nothing. If 'ExpectsException' than ILVerify is expected to throw an exception during verification, otherwise not
+        /// E.g.: SimpleAdd_Invalid_ExpectedNumericType_ExpectsException
+        /// </summary>      
+        public static IEnumerable<Object[]> GetMethodsWithInvalidIL()
+        {
+            var methodSelector = new Func<string[], EcmaMethod, EcmaMethodIL, TestCase>((mparams, method, methodIL) =>
+            {
+                if (mparams.Length == 4 && mparams[1].ToLower() == "invalid")
+                {
+                    var expectedErrors = mparams[2].Split('.');
+                    List<VerifierError> verificationErros = new List<VerifierError>();
```

Comment:
Nit: `var` would be more appropriate here to fit with the rest

BTW: We try to follow https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/coding-style.md where possible. It says this about the `var`: We only use `var` when it's obvious what the variable type is (i.e. `var stream = new FileStream(...)` not `var stream = OpenStandardInput()`). I know that the ILVerify sources are not fully conformant with this style ... follow it when in doubts or for large chunks of new code.